kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -3978,44 +3993,76 @@ public void hideAwayMessage(Message message) {
     }
 
     @Override
-    public void onAction(Context context, String action, Message message, Object object) {
+    public void onAction(Context context, String action, Message message, Object object, Map<String, Object> replyMetadata) {
         switch (action) {
             case AlRichMessage.SEND_GUEST_LIST:
                 List<ALGuestCountModel> guestCountModels = (List<ALGuestCountModel>) object;
-                sendGuestListMessage(guestCountModels);
+                sendGuestListMessage(guestCountModels, getStringMap(replyMetadata));
                 break;
 
             case AlRichMessage.SEND_HOTEL_RATING:
-                sendMessage((String) object);
+                sendMessage((String) object, getStringMap(replyMetadata));
                 break;
 
             case AlRichMessage.SEND_HOTEL_DETAILS:
-                sendHotelDetailMessage((AlHotelBookingModel) object);
+                sendHotelDetailMessage((AlHotelBookingModel) object, getStringMap(replyMetadata));
                 break;
 
             case AlRichMessage.SEND_ROOM_DETAILS_MESSAGE:
```

Comment:
At some point, we should remove these `Hotel` related cases and use the generic template Ids.